conda_smithy/templates/azure-pipelines-win.yml.tmpl

```diff
@@ -56,9 +56,9 @@ jobs:
 {%- endif %}
     - task: CondaEnvironment@1
       inputs:
-        packageSpecs: 'python=3.9 conda-build conda pip {{ BOA }}{{ " ".join(remote_ci_setup) }}' # Optional
-        installOptions: "-c conda-forge"
-        updateConda: true
+        packageSpecs: 'python=3.9 conda-build conda=4.12 pip {{ BOA }}{{ " ".join(remote_ci_setup) }}' # Optional
```

Comment:
Why is this necessary?

Reply:
Do you mean the general PR or this line? It's supposed to fix this error: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=513944&view=logs&jobId=171a126d-c574-5c8c-1269-ff3b989e923d&j=171a126d-c574-5c8c-1269-ff3b989e923d&t=b5c76c08-4f10-5244-840d-58e98b91712a

Reply:
this line

Reply:
https://gitter.im/conda-forge/conda-forge.github.io?at=6294e1824aa6c31dca2977b1

In the previous tests, it wasn't automatically downgraded and failed instead.

Reply:
I've simply pinned conda to 4.12 to update it (if necessary), but not beyond 4.12. 

The `updateConda: true` option does not take into account any additional arguments passed in the `installOptions` so we cannot restrict the action to `conda-forge` and it will update to 4.13 from defaults.

Later, conda cannot downgrade because I think the solver restricts it (but mamba & boa currently require 4.12).

Reply:
Please add any conda conditions to `conda-forge-ci-setup` package. We don't want to keep updating this in conda-smithy and rerendering every other feedstock

Reply:
this is fixing an issue that currently affects all feedstocks. 

It's not really possible otherwise to restrict conda to 4.13 since defaults is still available at this stage of the Azure Pipeline.
I think what you are suggesting is a bigger change.

Reply:
Or, if you prefer, I can remove the version pinning and just add the `--strict-channel-priority` and that should (hopefully) do the trick to only update to `conda 4.12` max.

Reply:
this is what the `updateConda: true` does: https://github.com/microsoft/azure-pipelines-tasks/blob/2135c795837e3692569b3baa4cccefc9f4c9b94d/Tasks/CondaEnvironmentV1/conda_internal.ts#L93-L102

Reply:
> Or, if you prefer, I can remove the version pinning and just add the --strict-channel-priority and that should (hopefully) do the trick to only update to conda 4.12 max.

Do this and also add the `conda=4.12`  to `conda-forge-ci-setup`. I don't understand why you think this wouldn't work.

Reply:
@isuruf you are referring to this line in the meta.yaml, right? https://github.com/conda-forge/conda-forge-ci-setup-feedstock/blob/c7a5453483f65d0e307249250bb180e033d41d4f/recipe/meta.yaml#L62

I am sure I can add the same constraint there.

Reply:
Btw. that constraint already exists in the `boa` package now (which is why we're seeing this error). conda is not "downgradeable" after an update.

Reply:
@wolfv, I'm only talking about this line as I mentioned previously. Not the other lines where install options and updateConda is changed.